vars/karmaExecuteTests.groovy

```diff
@@ -0,0 +1,73 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GitUtils
+import com.sap.piper.Utils
+
+import groovy.text.SimpleTemplateEngine
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'karmaExecuteTests'
+@Field Set GENERAL_CONFIG_KEYS = [
+    'containerPortMappings', //port mappings required for containers. This will only take effect inside a Kubernetes pod, format [[containerPort: 1111, hostPort: 1111]]
+    'dockerEnvVars', //envVars to be set in the execution container if required
+    'dockerImage', //Docker image for code execution
+    'dockerName', //name of the Docker container. If not on Kubernetes pod, this will define the network-alias to the NPM container and is thus required for accessing the server, example http://karma:9876 (default).
+    'dockerWorkspace', //user home directory for Docker execution. This will only take effect inside a Kubernetes pod.
+    'failOnError',
+    'installCommand',
+    'modules',
+    'runCommand',
+    'sidecarEnvVars', //envVars to be set in Selenium container if required
+    'sidecarImage', //image for Selenium execution which runs as sidecar to dockerImage
+    'sidecarName', //name of the Selenium container. If not on Kubernetes pod, this will define the network-alias to the Selenium container and is thus required for accessing the server, example http://selenium:4444 (default)
+    'sidecarVolumeBind', //volume bind. This will not take effect in Kubernetes pod.
+    'stashContent' //list of stash names which are required to be unstashed before test run
+]
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+void call(Map parameters = [:]) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        def script = parameters?.script ?: [commonPipelineEnvironment: commonPipelineEnvironment]
```

Comment:
Usage of a map containing the step specific CPE has been removed from all the steps with PR #305. We should try to not introduce it again.

There is now a Prerequisites class checking for script availability (it is a required parameter). For the time until we really enforce it, it is a better strategy to use `this` as script. `this` allows to access the commonPipelineEnvironment in exactly the same way like with the map, but in addition it is also possible to call other "steps", like `echo`, `sh` ...